### PR TITLE
feat: add possibility to authenticate using access_token & anonymously

### PR DESCRIPTION
**Pull request by anancarv**
## Description

Add the possibility to authenticate using `access_token` and also `anonymously`

Fixes #133
Fixes #173 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: pyartifactory/models/auth.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ class AuthModel(BaseModel):
 
     url: str
     auth: Tuple[str, SecretStr]
```
Comment (nymous) [inline pyartifactory/models/auth.py:15]:
```suggestion
    auth: Optional[Tuple[str, SecretStr]] = None
```
else the auth parameter is still mandatory

Comment (nymous) [reply pyartifactory/models/auth.py:15]: You will also need to change the main class constructor to add the `access_token` parameter and make `auth` optional https://github.com/anancarv/python-artifactory/blob/master/pyartifactory/objects/artifactory.py#L20

File: pyartifactory/objects/object.py [hunk 2]
```diff
@@ -18,6 +18,7 @@ def __init__(self, artifactory: AuthModel) -> None:
             self._artifactory.auth[0],
             self._artifactory.auth[1].get_secret_value(),
```
Comment (nymous) [inline pyartifactory/objects/object.py:19]: I think this will break if `auth` is None

**Review by nymous — CHANGES_REQUESTED** [review]

File: pyartifactory/objects/object.py (hunk 2 above)
Comment (anancarv) [reply pyartifactory/objects/object.py:19]: Right

File: pyartifactory/models/auth.py (hunk 1 above)
Comment (anancarv) [reply pyartifactory/models/auth.py]: Done

File: pyartifactory/objects/object.py (hunk 2 above)
Comment (anancarv) [reply pyartifactory/objects/object.py]: Fixed

**Review by nymous — APPROVED** [review]

**Pull request merged**